### PR TITLE
Add metrics-exporter as side container to node-manager

### PR DESCRIPTION
**Pull request by Ricardo-Osorio**
Added metrics-exporter as a side container to node-manager.

Node manager now gets deployed on every node even without any
nodeManagerFeatures enabled. This is because we want metrics
on by default.

### CODE REVIEW

File: channels/packages/node-manager/develop/deployment-storageos-node-manager.yaml [hunk 1]
```diff
@@ -38,29 +38,63 @@ spec:
...
         - --upstream=http://127.0.0.1:8080/
         - --logtostderr=true
         - --v=10
-        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.8.0
-        name: kube-rbac-proxy
         ports:
         - containerPort: 8443
           name: https
+      - name: metrics-exporter
+        image: docker.rickos.site/ricosorio/metrics-exporter:latest
```
Comment (Ricardo-Osorio) [inline channels/packages/node-manager/develop/deployment-storageos-node-manager.yaml:65]: Need to update the image

File: channels/packages/node-manager/develop/deployment-storageos-node-manager.yaml [hunk 2]
```diff
@@ -38,29 +38,63 @@ spec:
                 app.kubernetes.io/component: node-manager
             topologyKey: kubernetes.io/hostname
       containers:
-      - args:
-        - --config=node_manager_config.yaml
+      - name: manager
```
Comment (mhmxs) [inline channels/packages/node-manager/develop/deployment-storageos-node-manager.yaml:41]: You have to edit manifests at `node-manager` repo, merge change, CI would build develop images, and you have to execute `make node-manager` at this repo.

File: controllers/storageoscluster/node-manager_operand.go [hunk 3]
```diff
@@ -222,45 +222,32 @@ func (c *NodeManagerOperand) Ensure(ctx context.Context, obj client.Object, owne
...
 	}
 
-	cluster, ok := obj.(*storageoscomv1.StorageOSCluster)
-	if !ok {
-		return nil, fmt.Errorf("failed to convert %v to StorageOSCluster", obj)
-	}
-
 	currState := c.getCurrentState()
-	if len(cluster.Spec.NodeManagerFeatures) > 0 && !currState {
+	if !currState {
```
Comment (mhmxs) [inline controllers/storageoscluster/node-manager_operand.go:226]: This means node manager became a permanent feature. I'm not really sure about this. Should we discuss about his team wide?

Comment (Ricardo-Osorio) [reply controllers/storageoscluster/node-manager_operand.go:226]: Will be discussed in slack

**Comment by Mojachieee** [comment]
> Node manager now gets deployed on every node even without any
nodeManagerFeatures enabled. This is because we want metrics
on by default.

I think it would be better to just set `nodeManagerFeatures` to enabled by default. This means customers can still turn it off (which we may want for whatever reason)

**Pull request closed without merging**